docs/source/User-Guide.rst

```diff
@@ -242,3 +242,43 @@ To read it from file
     >>> with open('configspace.json', 'r') as fh:
     ...     json_string = fh.read()
     ...     restored_conf = json.read(json_string)
+
+
+
+4th Example: Placing priors on the hyperparameters
+-------------------------
+
+If you want to conduct black-box optimization in SMAC, and you have prior knowledge about the which regions of the search space are more likely to contain the optimum, you may include this knowledge when designing the configuration space. More specifically, you place prior distributions over the optimum on the parameters, either by a (log)-normal or (log)-Beta distribution.
+
+Consider the case of optimizing the accuracy of an MLP with three hyperparameters: learning rate [1e-5, 1e-1], dropout [0, 0.99] and activation {Tanh, ReLU}. From prior experience, you believe the optimal learning rate to be around 1e-3, a good dropout to be around 0.25, and the optimal activation function to be ReLU about 80% of the time. This can be represented accordingly:
+
+>>> import numpy as np
+>>> import ConfigSpace.hyperparameters as CSH
+>>> from ConfigSpace.configuration_space import ConfigurationSpace
+>>> # convert 10 log to natural log for learning rate, mean 1e-3
+>>> logmean = np.log(np.power(10.0, -3))
+>>> # two standard deviations on either side of the mean to cover the search space
+>>> logstd = np.log(10.0) 
+>>> learning_rate = CSH.NormalFloatHyperparameter(name='learning_rate', lower=1e-5, upper=1e-1, default_value=1e-3, mu=logmean, sigma=logstd, log=True)
```

Comment:
I agree. This is exactly what I'm trying to look at right now. Also, there's some silly stuff going on with the default values (if none is given) in that it assigns
`self.default_value=self.mu` 
which is often going to be out of bounds for the log case (and doesn't correspond to the default value the user had in mind.

What we have now corresponds to when an actual log-normal distribution is defined, but it's not optimal for intuition.

Reply:
Okay, that makes sense but yeah as you point out it's not intuitive, I made a note at the bottom of this review and I think `self.default_value` is something that could be wrapper in a property to ensure it's always giving what's intended.

Reply:
We can definitely remove it, but I unfortunately think this one becomes a bit insensible if we do! I would rather add another parameter in the example, which I can look to do.

Reply:
@eddiebergman Okay, so now this is what happens:

You still define the mean and standard deviation in this slightly awkward way. However, I do think that this is best, as you don't have to guess (but just go to Wikipedia) if you don't know your lognormal distribution by heart. If you do not specify the default value as above, it would still have 1e-3 as the default, and say as much.

However, mu and sigma would have the slightly awkward values of -6.9 and 2.3 which, once again, is mathematically correct. An option would be to add some sort of parenthesis to the repr() of the parameter, but this is probably not going to look great.